File: c/gpt/p106.c
#include <stdio.h>
#include <string.h>
#include <sys/stat.h>
#include <unistd.h>

/* p106.c
   Write a super-secret string to a file, set permissions to owner read-only (0400), print filename.
*/
int main(void) {
    const char *secret = "super secret info: do not share";
    const char *fname = "super_secret.txt";
    FILE *f = fopen(fname, "w");
    if (!f) { perror("fopen"); return 1; }
    fprintf(f, "%s\n", secret);
    fclose(f);
    if (chmod(fname, S_IRUSR) != 0) perror("chmod");
    printf("%s\n", fname);
    return 0;
}

File: java/gpt/p106.java
import java.io.*;

// p106.java - create a file containing super secret info, set owner read-only (best-effort), print filename
public class p106 {
    public static void main(String[] args) {
        String fname = "super_secret.txt";
        String secret = "super secret info: do not share";
        try (FileWriter fw = new FileWriter(fname)) { fw.write(secret + "\n"); }
        catch (IOException e) { e.printStackTrace(); System.exit(1); }
        File f = new File(fname);
        f.setReadable(true, true);
        f.setWritable(false, true);
        System.out.println(fname);
    }
}
